src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -299,22 +342,24 @@ abstract class AbstractCheckTask(
 
     override fun onComplete() {
         super.onComplete()
-        end("Message stream is completed")
+        end(executionState, "Message stream is completed")
     }
 
     /**
      * Prepare the root event or children events for publication.
      * This method is invoked in [State.PUBLISHED] state.
      */
-    protected open fun completeEvent(canceled: Boolean) {}
+    protected open fun completeEvent(taskState: State) {}
 
     /**
      * Publishes the event to [eventBatchRouter].
      */
     private fun publishEvent() {
         val prevState = taskState.getAndSet(State.PUBLISHED)
         if (prevState != State.PUBLISHED) {
-            completeEvent(prevState == State.TIMEOUT)
+            if (!untrusted || isIgnoreUntrustedFlag()) {
```

Comment:
Why do we ignore the untrusted flag in some cases?

Reply:
If we passed untrusted flag, some pre initialized data in method `onStart` will be ignored due to `onSubscribe` method not being called, so we have to check that